plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithMultinodeQueryRunner.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.cache;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.HiveQueryRunner;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static io.trino.tpch.TpchTable.NATION;
+
+@Test(singleThreaded = true)
+public class TestCachingHiveMetastoreWithMultinodeQueryRunner
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HiveQueryRunner.builder()
+                .setHiveProperties(ImmutableMap.of("hive.metastore-cache-ttl", "60m"))
+                // only so that tpch schema is created
+                .setInitialTables(ImmutableList.of(NATION))
+                // multinode with dedicated worker and coordinator
+                .setNodeCount(2)
+                .setCoordinatorProperties(ImmutableMap.of("node-scheduler.include-coordinator", "false"))
```

Comment:
This is necessary to make the reproduction of the original problem deterministic

```
// Exclude coordinator from workers to make testPartitionAppend deterministically reproduce the original problem
```